### DISCUSSION TITLE
Rotate log data?

### DISCUSSION THREAD

**Discussion by ekandreas** [spatie/laravel-activitylog#1049] [General]
Is it possible to rotate the logged data from the database into a file eg,
Everything from logged to a monolog formatted file for storage, then truncate the activity log table.
How would you approach this challange?

**Comment by Gummibeer** [comment]
You can take the clean up command as a starting point.
https://github.com/spatie/laravel-activitylog/blob/main/src/CleanActivitylogCommand.php

How you format each record is up to you. But at the end it will be:
1) retrieve old records
2) loop each record
3) print each record to file
4) delete each printed record
